app/Http/Controllers/API/CategoryController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Http\Controllers\API;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\API\CategoryController\CategoryRequest;
+
+use App\Http\Resources\CategoryResource;
+use App\Models\Category;
+use stdClass;
+
+class CategoryController extends Controller
+{
+    private int $indexStatusCode = 200;
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function index(): \Illuminate\Http\JsonResponse
+    {
+        return (CategoryResource::collection(Category::whereNull("parent_id")->with("children")->latest()->get()))
+                    ->response()
+                    ->setStatusCode($this->indexStatusCode);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param CategoryRequest $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function store(CategoryRequest $request): \Illuminate\Http\JsonResponse
+    {
+        Category::create($request->validated());
+        $this->indexStatusCode = 201;
+        return $this->index();
```

Comment:
never ever do not use any of controller methods in other methods :)
make a service instead of this.